### PR TITLE
Add Back PR Template

### PR DESCRIPTION
**Pull request by yangw1234**
## Description

This PR adds a pull request template.

## Motivation and Context

This is a good practice to add a structured short description of a submitted PR. This way the reviewer and future developers can better understand the changes and the context of the changes.

## API Usage or Code Design.

The proposed PR template is presented as in this description.

## Related Link or Issue

N/A

## How was this PR tested?
N/A

- [ ] jenkins passed (please provide link):

- [ ] documentation build passed (please provide link): 

## License & Dependency
N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## What changes were proposed in this pull request?
+
+(Please fill in changes proposed in this patch)
```
Comment (jason-dai) [inline .github/PULL_REQUEST_TEMPLATE:3]: In this section, we need:
1) What is the change?
2) Why the change?
3) Summary of the change (interface and short design) - needed unless we have a gthub issue for the design below

**Comment by jason-dai** [comment]
> ## What changes were proposed in this pull request?
> This PR adds back the pull request template.
> 
> ## How was this patch tested?
> No need to test.
> 
> ## Related links or issues (optional)
> N/A

1. What is the change?
2. Why the change?
3. Summary of the change (interface and design)
    (Not needed if section 4 is provided)
4. Link for the design
    (Not needed if section 3 is provided)
6. How to test?

**Comment by qiyuangong** [comment]
We can add some options for PR type, (doc, scala, python). This may leads to doc review, scala test, python test.

**Comment by yushan111** [comment]
Shall we add different templates for code and documents?

**Comment by jason-dai** [comment]
> Shall we add different templates for code and documents?

I think we can make it simpler for small changes - e.g., if there are no more than 3 source file changes or 50 line-of-code changes, then maybe the PR just needs a simple description?

**Comment by TheaperDeng** [comment]
template might teach our developers on what circumstances their PR will affect the document, if applied they should generate a preview document and put the link in the description.

**Comment by qiuxin2012** [comment]
need `Related issue` (optional)

**Comment by sgwhat** [comment]
Maybe add a template for users to learn how to use the API? Like if this is a new data input API, we could provide a simple description of how to process data and write code using this API. 

Also, maybe add a link to the related example or ut if they are not included in this PR?

**Comment by zhentaocc** [comment]
Can we mark these items as required or optional?  
For some PRs that only fix a small bug with only a few changes, is it necessary to fully document the PR description?

What is the difference between `What changes were proposed in this pull request` and `briefly describe the changes`? The title seems not that clear.

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## What changes were proposed in this pull request?
+
+## Why are the changes needed?
+
+## Please briefly describe the changes such as API changes and code design.
+
+(This section is needed unless a github issue describing the design is provide below)
+
```
Comment (liu-shaojun) [inline .github/PULL_REQUEST_TEMPLATE:8]: Would it be possible to update the title **"Please briefly describe the changes such as API changes and code design."** to a more general and brief title such as **"Description"**?
```
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
```
Would it be better to put the `## Description` in the first section? And comment out the descriptive text using `<!--- just --->`

**Comment by cyita** [comment]
Add `Jenkins or read the docs link(optional)`?

**Comment by glorysdj** [comment]
May need a question for license and BOM check.
Did you bring in any new dependencies for java/scala or python?
IF yes. What are they? What are their licenses? And the Copyright?
Did you pass the license scan? Did you update the third party license, copyright and bom file?

**Comment by jason-dai** [comment]
> template might teach our developers on what circumstances their PR will affect the document, if applied they should generate a preview document and put the link in the description.
> 
> * documentation build checklist:
>   
>   1. No new `import` are involved on file headers.
>   2. No file path is changed.
>   3. No files in `/docs` is changed.
>   
>   
>   * [ ]  document pass (link is needed if any of above is not met):

How does one know if the document passes or not?

**Comment by TheaperDeng** [comment]
> > template might teach our developers on what circumstances their PR will affect the document, if applied they should generate a preview document and put the link in the description.
> > 
> > * documentation build checklist:
> >   
> >   1. No new `import` are involved on file headers.
> >   2. No file path is changed.
> >   3. No files in `/docs` is changed.
> >   
> >   
> >   * [ ]   document pass (link is needed if any of above is not met):
> 
> How does one know if the document passes or not?

run this readthedoc pr check http://10.112.231.51:18888/view/BigDL-2.0-PR/job/BigDL-PR-Readthedocs-check/ will do

**Comment by yangw1234** [comment]
> > > template might teach our developers on what circumstances their PR will affect the document, if applied they should generate a preview document and put the link in the description.
> > > 
> > > * documentation build checklist:
> > >   
> > >   1. No new `import` are involved on file headers.
> > >   2. No file path is changed.
> > >   3. No files in `/docs` is changed.
> > >   
> > >   
> > >   * [ ]    document pass (link is needed if any of above is not met):
> > 
> > 
> > How does one know if the document passes or not?
> 
> run this readthedoc pr check http://10.112.231.51:18888/view/BigDL-2.0-PR/job/BigDL-PR-Readthedocs-check/ will do

Does it make sense to add the check to github action? Or including it in all components' jenkins job?

**Comment by yangw1234** [comment]
> > > template might teach our developers on what circumstances their PR will affect the document, if applied they should generate a preview document and put the link in the description.
> > > 
> > > * documentation build checklist:
> > >   
> > >   1. No new `import` are involved on file headers.
> > >   2. No file path is changed.
> > >   3. No files in `/docs` is changed.
> > >   
> > >   
> > >   * [ ]    document pass (link is needed if any of above is not met):
> > 
> > 
> > How does one know if the document passes or not?
> 
> run this readthedoc pr check http://10.112.231.51:18888/view/BigDL-2.0-PR/job/BigDL-PR-Readthedocs-check/ will do

Does it make sense to add the check to github action? Or including it in all components' jenkins job?

**Comment by yangw1234** [comment]
Is it ok if we first merge this PR if the current content is OK to everybody? @jason-dai 

I think the remaining issues are:
1. whether to add a checklist regarding documentation build (@TheaperDeng )
2. whether to ask the author to specify the label (@shane-huang)

We can move the discussion to dedicated issues.

**Comment by TheaperDeng** [comment]
> Is it ok if we first merge this PR if the current content is OK to everybody? And address the remaining issues in follow-up PRs. @jason-dai
> 
> I think the remaining issues are:
> 
>     1. whether to add a checklist regarding documentation build (@TheaperDeng )
> 
>     2. whether to ask the author to specify the label (@shane-huang)
> 
> 
> We can move the discussion to dedicated issues.

I think we can leave document checklist and forced jenkins job a TODO. Since the jenkins job is not that ready (take too much time and can not catch all document generation issues).
